### ISSUE TITLE
The Google Sheets v3 API will be shut down on March 3, 2020

### ISSUE THREAD

**Issue by huan** [tadam313/sheet-db#7]
> The Google Sheets v3 API will be shut down on March 3, 2020. Follow this migration guide to update your applications so that they use the v4 API. Learn more

https://developers.google.com/sheets/api/v3/authorize

It seems that we need to upgrade our API to v4 before March 2020.

### Update Sep 2020

Here's a great google spreadsheet API module with v4 which we can learn from: [Google Sheets API (v4) wrapper for Node.js](https://github.com/theoephraim/node-google-spreadsheet)

**Comment by tadam313** [comment]
indeed, I reviewed this previously but didn't have time to do the actual change. One slight issue I could think of is according to my understanding v4 API dropped support of SQ (see in this section: https://developers.google.com/sheets/api/guides/migration#retrieve_row_data) which means we need to run the filtering query entirely on client side until they add support for that. This will affect performance but I can't see other choice right now.

Would you be able to help me out here with a PR? We have unit tests so hopefully it won't be that hard to change the API. We can even drop versioning (like keeping api files in v3, v4 folders) I don't think we're ever gonna support multiple APIs at once.

**Comment by huan** [comment]
I'm working with Google API recently, and I hope I will get some time to working on our repo in the next weeks.

Here's a great google spreadsheet API module with v4 which we can learn from: [Google Sheets API (v4) wrapper for Node.js](https://github.com/theoephraim/node-google-spreadsheet)
